versions/1/data_stream/fields/fields.spec.yml

```diff
@@ -6,66 +6,82 @@ spec:
   type: array
   items:
     type: object
+    # TODO: We should disable additionalProperties. There are properties used in
+    # elastic/integrations that have no purpose or are typos.
     additionalProperties: true
     properties:
       name:
-        description: Name of field
-        type: string
-      title:
-        description: Title of field
+        description: >
+          Name of field. Names containing dots are automatically split into
+          sub-fields.
         type: string
+        # TODO: Should we allow wildcards if they are not implemented in Fleet?
+        pattern: '^[_@A-Za-z0-9]+(\.[_@A-Za-z0-9]+)*$'
       type:
         description: Datatype of field
         type: string
         enum:
```

Comment:
No changes to the contents of this list were made. I just sorted it to make identifying missing types a little easier.

This is not a complete list, but I noticed `geo_shape` and `dense_vector` are supported in ES but not in packages.

Reply:
If these fields are not used in current packages and is not supported by Fleet, I would say to add them later when needed.